### ISSUE TITLE
[BUG]: App Crashes when we call JitsiMeet.closeMeeting()

### ISSUE THREAD

**Issue by NifalNizar** [gunschu/jitsi_meet#187]
**Platform: Android
Device Physical or Simulator:Physical
Device Model: REDMI NOTE 9 PRO
Pluggin: jitsi_meet: ^2.0.0**

E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.getJitsiView(JitsiMeetActivity.java:110)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.leave(JitsiMeetActivity.java:126)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.finish(JitsiMeetActivity.java:99)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.onConferenceTerminated(JitsiMeetActivity.java:223)
E/JitsiMeetSDK(30570): 	at com.gunschu.jitsi_meet.JitsiMeetPluginActivity.onConferenceTerminated(JitsiMeetPluginActivity.kt:88)
E/JitsiMeetSDK(30570): 	... 11 more
E/JitsiMeetSDK(30570):
E/JitsiMeetSDK(30570): java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.ListenerUtils.runListenerMethodOnUiThread(ListenerUtils.java:139)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.ListenerUtils.access$000(ListenerUtils.java:35)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.ListenerUtils$1.run(ListenerUtils.java:114)
E/JitsiMeetSDK(30570): 	at android.os.Handler.handleCallback(Handler.java:883)
E/JitsiMeetSDK(30570): 	at android.os.Handler.dispatchMessage(Handler.java:100)
E/JitsiMeetSDK(30570): 	at android.os.Looper.loop(Looper.java:224)
E/JitsiMeetSDK(30570): 	at android.app.ActivityThread.main(ActivityThread.java:7562)
E/JitsiMeetSDK(30570): 	at java.lang.reflect.Method.invoke(Native Method)
E/JitsiMeetSDK(30570): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
E/JitsiMeetSDK(30570): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
E/JitsiMeetSDK(30570): Caused by: java.lang.reflect.InvocationTargetException
E/JitsiMeetSDK(30570): 	at java.lang.reflect.Method.invoke(Native Method)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.ListenerUtils.runListenerMethodOnUiThread(ListenerUtils.java:135)
E/JitsiMeetSDK(30570): 	... 9 more
E/JitsiMeetSDK(30570): **Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.jitsi.meet.sdk.JitsiMeetView org.jitsi.meet.sdk.JitsiMeetFragment.getJitsiView()' on a null object reference**
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.getJitsiView(JitsiMeetActivity.java:110)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.leave(JitsiMeetActivity.java:126)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.finish(JitsiMeetActivity.java:99)
E/JitsiMeetSDK(30570): 	at org.jitsi.meet.sdk.JitsiMeetActivity.onConferenceTerminated(JitsiMeetActivity.java:223)
E/JitsiMeetSDK(30570): 	at com.gunschu.jitsi_meet.JitsiMeetPluginActivity.onConferenceTerminated(JitsiMeetPluginActivity.kt:88)
E/JitsiMeetSDK(30570): 	... 11 more
I/Process (30570): Sending signal. PID: 30570 SIG: 9
Lost connection to device.

**Comment by tuantvu** [comment]
@NifalNizar Could you add your code snippets so we can recreate? Thanks

**Comment by NifalNizar** [comment]
**Im trying to close the meeting manually through function JitsiMeet.closeMeeting();**

```
void _loadJitsi() async {
    try {
      print('-----------_loadJitsi---------------');
      // Specify your own hosted server. Must be a valid absolute URL of the format <scheme>://<host>[/path],
      // i.e. https://someHost.com. Defaults to Jitsi Meet's servers.
      String serverUrl = "https://telecon.medoment.com/";

      Codec<String, String> stringToBase64 = utf8.fuse(base64);
      String room = stringToBase64.encode(appointmentId.toString());
      room = room.replaceAll(new RegExp('/[^0-9a-zA-Z]/g'), "_");

      String subject = "Call with " + widget.teleconInformation.doctorName;

      // Enable or disable any feature flag here
      // If feature flag are not provided, default values will be used
      // Full list of feature flags (and defaults) available in the README
      FeatureFlag featureFlag = FeatureFlag();
      featureFlag.welcomePageEnabled = false;
      featureFlag.addPeopleEnabled = false;
      featureFlag.inviteEnabled = false;

      // Here is an example, disabling features for each platform
      if (Platform.isAndroid) {
        // Disable ConnectionService usage on Android to avoid issues (see README)
        featureFlag.callIntegrationEnabled = false;
      } else if (Platform.isIOS) {
        // Disable PIP on iOS as it looks weird
        featureFlag.pipEnabled = false;
      }

      // Define meetings options here
      var options = JitsiMeetingOptions()
        ..room = widget.teleconInformation.meetingRoom
        ..serverURL = serverUrl
        ..subject = subject
        ..userDisplayName = widget.teleconInformation.patientName + " (Patient)"
        ..userEmail =
            widget.teleconInformation.meetingCode + "@telecon.medoment.com"
        ..audioOnly = !widget.teleconInformation.isVideo
        ..audioMuted = !widget.teleconInformation.isAudio
        ..videoMuted = !widget.teleconInformation.isVideo
        ..featureFlag = featureFlag;

      print("JitsiMeetingOptions: $options");
      await JitsiMeet.joinMeeting(options);
      isJitsiConnected = true;
      // JitsiMeet.addListener(JitsiMeetingListener(
      //   onConferenceTerminated: _onConferenceTerminated,
      // ));
    } catch (error) {
      print("-----------------_loadJitsi ----------error: $error");
      utility.showSomethingWentWrong(context, local);
    }
  }
```

```
void _endJitsiConnection() {
  JitsiMeet.closeMeeting();
}
```

**Comment by NifalNizar** [comment]
@tuantvu any updates on this?

https://github.com/jitsi/jitsi-meet/pull/7961

**Comment by Yogesh-333** [comment]
Any update or solution ?

**Comment by IggyBlob** [comment]
This prevents us from releasing our Android app.... Any updates?

**Comment by NifalNizar** [comment]
still no reply from the team, even Im unable to release my app.

**Comment by tuantvu** [comment]
Hi all, we're working on a big update for web and then null safety next. If you're able to troubleshoot and create a pull request to help us, that'd be great.

**Comment by IggyBlob** [comment]
@tuantvu I created PR #225 that updates the Jitsi Meet SDK to 3.2.0, which includes the missing null checks to prevent such NPEs.

@NifalNizar In the meantime, as a workaround, you may wanna use [my fork](https://github.com/IggyBlob/jitsi_meet) like this in your `pubspec.yaml`:

```
dependencies:
  flutter:
    sdk: flutter
  jitsi_meet:
    git:
      url: https://github.com/IggyBlob/jitsi_meet
```

**Comment by tuantvu** [comment]
Pulled @IggyBlob's PR and published version 2.1.0. @NifalNizar, please test and confirm this helps.
